### DISCUSSION TITLE
Memory usage

### DISCUSSION THREAD

**Discussion by glezmartin** [online-ml/river#922] [Q&A]
Hi everyone,

I am implementing an online algorithm based on prototypes. The algorithm passes the check_estimator test without problems but I have a doubt about how to allow the evaluation functions to measure the memory usage by the algorithm. As far as I have been able to see the Base class has the _memory_usage method that allows the _progressive_validation function to show it on screen. The algorithm I have implemented inherits from the Classifier class which should inherit that method from Base. However when I perform the evaluation of the algorithm it breaks. What am I doing wrong? Is it enough that the estimator is a child class of Classifier?

Thank you!

**Comment by MaxHalford** [comment]
Hey! What you're doing sounds right. Could you provide us with a stack trace of the code that errored? :)

**Reply by glezmartin** [reply]
Hello! There is no stack trace the program is killed.

**Reply by MaxHalford** [reply]
Ok then, can you at least share a reproducible piece of code?
